### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by YuliiaSaliuta**

### CODE REVIEW

**Review by Naz-iv — APPROVED** [review]
well done

File: app/main.py [hunk 1]
```diff
@@ -2,24 +2,69 @@
...
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+
+        validator = self.limitation_class(
+            visitor.age,
+            visitor.weight,
+            visitor.height
+        )
+        return all(field for field in validator.__dict__.values())
```
Comment (LiudmylaKulinchenko) [inline app/main.py]: `Create can_access method that takes instance of Visitor class and returns if the visitor can use the slide.` -- you should have this logic here, not in setter

File: app/main.py [hunk 2]
```diff
@@ -2,24 +2,69 @@
...
+
+    def __get__(self, instance: object, owner: type) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+
+        if value in range(self.min_amount, self.max_amount + 1):
+            setattr(instance, self.protected_name, value)
+        else:
+            setattr(instance, self.protected_name, False)
```
Comment (LiudmylaKulinchenko) [inline app/main.py]: raise a ValueError if value is in wrong range

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -2,24 +2,67 @@
...
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"Value {value} not in range")
```
Comment (LiudmylaKulinchenko) [inline app/main.py]: Better to specify what a range is here, because it is not really clear now

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -2,24 +2,67 @@
...
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"Value {value} not in range for this slide")
```
Comment (LiudmylaKulinchenko) [inline app/main.py:19]: You can also specify min and max amount

**Review by LiudmylaKulinchenko — APPROVED** [review]
Great! 😋

**Pull request closed without merging**